tool/release/createnotes/Note.kt

```diff
@@ -0,0 +1,145 @@
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelFeature
+import com.vaticle.dependencies.tool.release.createnotes.Constant.github
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelPrefix
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelRefactor
+
+class Note {
+    enum class Type { FEATURE, BUG, REFACTOR, OTHER }
+
+    private val title: String
+    private val goal: String?
+    private val type: Type
+
+    private constructor(title: String, goal: String?, type: Type) {
+        this.title = title
+        this.goal = goal
+        this.type = type
+    }
+
+    fun toMarkdown():  String {
+        return "- **$title**\n  ${goal ?: ""}"
+    }
+
+    companion object {
+        fun fromGithubPR(pr: JsonObject): Note {
+            return Note(
+                title = pr.get("title").asString(),
+                goal = getPRGoal(pr.get("body").asString()),
+                type = getPRType(pr.get("labels").asArray())
+            )
+        }
+
+        private fun getPRType(labels: JsonArray): Type {
+            val types = labels
+                .map { e -> e.asObject().get("name").asString() }
+                .filter { e -> e.startsWith(labelPrefix) }
+            val type =
+                when {
+                    types.contains(labelFeature) -> Type.FEATURE
+                    types.contains(labelBug) -> Type.BUG
+                    types.contains(labelRefactor) -> Type.REFACTOR
+                    else -> Type.OTHER
+                }
+            return type
+        }
+
+        private fun getPRGoal(description: String): String {
+            val goal = StringBuilder()
+            var header = 0
+            for (line in description.lines()) {
+                if (line.startsWith("##")) {
+                    header += 1
+                } else if (header == 1) {
+                    goal.append(line)
+                } else if (header > 1) {
+                    break
+                }
+            }
+            return goal.toString()
+        }
+
+        fun fromGithubCommit(commit: JsonObject): Note {
+            return Note(commit.get("message").asString().lines().first(), null, Type.OTHER)
+        }
+
+        fun toMarkdown(notes: List<Note>): String {
+            val features = mutableListOf<Note>()
+            val bugs = mutableListOf<Note>()
+            val refactors = mutableListOf<Note>()
+            val others = mutableListOf<Note>()
+
+            notes.forEach { note ->
+                when (note.type) {
+                    Type.FEATURE -> features.add(note)
+                    Type.BUG -> bugs.add(note)
+                    Type.REFACTOR -> refactors.add(note)
+                    else -> others.add(note)
+                }
+            }
+
+            return """
+## New Features
+${features.map { e -> e.toMarkdown() }.joinToString("\n")}
+
+## Bugs Fixed
+${bugs.map { e -> e.toMarkdown() }.joinToString("\n")}
```

Comment:
@lolski can we use method reference here?

```suggestion
${bugs.map(Note::toMarkdown() }.joinToString("\n")}
```

Reply:
Fixed in 3be6d1dd53f720c2490e05d2d4d2e77b078fdac1.